device/src/web.ts

```diff
@@ -106,6 +107,31 @@ export class DeviceWeb extends WebPlugin implements DevicePlugin {
       uaFields.operatingSystem = 'unknown';
     }
 
+    // Check for browsers based on non-standard javascript apis, not user agent
+    const isFirefox = !!(window as any).InstallTrigger;
+    const isSafari = !!(window as any).ApplePaySession;
+    const isChrome = !!(window as any).chrome;
```

Comment:
It is preferred to use module augmentation on global types instead of casting them to `any`. You can do this by adding to the `declare global` at the top. Something like this:

```ts
declare global {
  interface Window {
    InstallTrigger?: any;
    ApplePaySession?: any;
    chrome?: any;
  }

  ...
}
```